pubtools/pulplib/_impl/client/client.py

```diff
@@ -363,3 +410,35 @@ def _delete_upload_request(self, upload_id):
 
         LOG.debug("Deleting upload request %s", upload_id)
         return self._request_executor.submit(self._do_request, method="DELETE", url=url)
+
+    def _do_get_maintenance(self):
+        def map_404_to_none(response):
+            """Given a future, if it resolved with a 404 error then return a Future[None],
+            else return original future"""
+            out = Future()
+
+            def handle_response(ft):
+                exception = ft.exception()
+                if (
+                    hasattr(exception, "response")
+                    and exception.response.status_code == 404
+                ):
+                    # failed with 404 => returned future should return None
+                    out.set_result(None)
+                elif exception:
+                    # failed with other exception => returned future should raise that exception
+                    out.set_exception(exception)
```

Comment:
Now this seems to be uncovered, so there should be a case where fetching the JSON file fails with (e.g.) 500 error.